apps/files/src/actions/moveOrCopyAction.ts

```diff
@@ -86,33 +109,43 @@ export const handleCopyMoveNodeTo = async (node: Node, destination: Folder, meth
 	 * node: /foo/bar/file.txt -> path = /foo/bar
 	 * destination: /foo
 	 * Allow move of /foo does not start with /foo/bar so allow
+	 * But allow copy a file to the same directory
 	 */
-	if (destination.path.startsWith(node.path)) {
+	if (destination.path.startsWith(node.path) && destination.path !== node.path) {
```

Comment:
Does it prevent moving a folder into itself? Not sure how it could happen, but still.
```js
destination.path = "/a/b"
node.path = "/a/b"

destination.path.startsWith(node.path) === true
(destination.path !== node.path) === false
```

I think the following would work:
```suggestion
	if (destination.path.startsWith(node.path) && (destination.path !== node.path || node.type === Folder)) {
```

But warning, I might simply be confused.

Reply:
No the whole `&& ...` can be dropped.

Because if your target folder is `/foo/bar` then ofcourse I can not move `/foo` into it, but I can also not move `/foo/bar` into it, and I am even not allowed to copy it into itself, because if both have the same path I try:
`/foo/bar` -> `/foo/bar` leading to `/foo/bar/bar` which is also not allowed.

To clone that directory you would need to pass `/foo` as the destination and then the check is ok as it was.